### PR TITLE
Rescue "invalid byte sequence in UTF-8", with a custom message

### PR DESCRIPTION
**Pull request by mvidner**
[bsc#992821](https://bugzilla.suse.com/show_bug.cgi?id=992821)

### CODE REVIEW

**Comment by mvidner** [comment]
- [ ] fill in the number
- [ ] version + changelog

File: src/ruby/yast/wfm.rb [hunk 1]
```diff
@@ -194,9 +194,17 @@ def self.ask_to_run_debugger?
     # @param [Exception] e the caught exception
     # @return [String] human readable exception description
     def self.internal_error_msg(e)
-      "Internal error. Please report a bug report with logs.\n" \
-        "Details: #{e.message}\n" \
-        "Caller:  #{e.backtrace.first}"
+      msg = "Internal error. Please report a bug report with logs.\n"
+
+      if e.is_a?(ArgumentError) && e.message =~ /invalid byte sequence in UTF-8/
+        msg += "A string was encountered that is not valid in UTF-8.\n" \
```
Comment (jreidinger) [inline src/ruby/yast/wfm.rb]: does it make sense to have message prefixed by internal error problem? I think different start of message will be better if is it specific for this issue.

File: src/ruby/yast/wfm.rb [hunk 2]
```diff
@@ -194,9 +194,17 @@ def self.ask_to_run_debugger?
...
+               "The system encoding is #{Encoding.locale_charmap.inspect}.\n" \
+               "Refer to https://www.suse.com/support/kb/doc?id=1234567.\n\n"
+        # FIXME: fill in TID number
+      end
+
+      msg + "Details: #{e.message}\n" \
+            "Caller:  #{e.backtrace.first}"
```
Comment (jreidinger) [inline src/ruby/yast/wfm.rb]: does it make sense this for Argument error problem? especially when we know message?

File: src/ruby/yast/wfm.rb (hunk 1 above)
Comment (mvidner) [inline src/ruby/yast/wfm.rb]: In the end it still is an internal error and we still want a bug report with logs, so it is better to keep it.

File: src/ruby/yast/wfm.rb (hunk 2 above)
Comment (mvidner) [inline src/ruby/yast/wfm.rb]: Yes, it is redundant. IMHO it is better to keep it than complicate the rescue handler even further.

**Comment by jreidinger** [comment]
LGTM

**Comment by mvidner** [comment]
(Jenkins build fails because OBS cannot see the IBS repo, then GitHub shows it as still pending. Weird)

**Pull request merged**